src/NuGet.Core/NuGet.Frameworks/CompatibilityProvider.cs

```diff
@@ -48,11 +48,13 @@ public bool IsCompatible(NuGetFramework target, NuGetFramework candidate)
             // check the cache for a solution
             var cacheKey = new CompatibilityCacheKey(target, candidate);
 
-            bool? result = _cache.GetOrAdd(cacheKey, (Func<CompatibilityCacheKey, bool>)((key)
-                =>
-            { return IsCompatibleCore(target, candidate) == true; }));
+            if (!_cache.TryGetValue(cacheKey, out bool result))
+            {
+                result = IsCompatibleCore(target, candidate) == true;
```

Comment:
``` suggestion
                result = IsCompatibleCore(target, candidate).Value;
```

It looks like `IsCompatibleCore` returns a `nullable bool` there is no code path that returns a `null` value in it. Hence I thought `result = IsCompatibleCore(target, candidate).Value;` would be sufficient.